mkosi/config.py

```diff
@@ -2107,6 +2164,16 @@ def outputs(self) -> list[str]:
             self.output_tar,
         ]
 
+    @property
+    def build_subdir(self) -> Path:
+        assert self.build_dir
+        subdir = self.expand_key_specifiers(self.build_key)
+
+        if subdir == "-":
+            return self.build_dir
+
+        return self.build_dir / subdir
```

Comment:
Do we actually need the extra semantic for `-` here? If subdir is the empty string, this would naturally fall back to the build directory and we don't prevent an empty build key, or do we?

Reply:
Assigning the empty string to a setting resets it back to the default value. That's why I added this.